src/pybind/mgr/volumes/fs/operations/versions/subvolume_v1.py

```diff
@@ -768,7 +768,10 @@ def get_pending_clones(self, snapname):
 
         for track_id in pending_track_id_list:
             try:
-                link_path = self.fs.readlink(os.path.join(index_path, track_id), 4096)
+                path = os.path.join(index_path, track_id)
+                link_path = self.fs.readlink(path, 4096)
+                if not os.path.exists(link_path):
+                    self.fs.unlink(path)
```

Comment:
We should ignore any kind of exception that could be raised by `fs.unlink` by logging a warning if it's important. Since this is an optimization, it should not fail the main operation i.e., snapshot info